pkg/test/live/runner.go

```diff
@@ -82,6 +85,22 @@ func (r *Runner) Run(t *testing.T) {
 	}
 }
 
+func (r *Runner) RunPreApply(t *testing.T) {
+	preApplyDir := filepath.Join(r.Path, "pre-apply")
+	fi, err := os.Stat(preApplyDir)
+	if err != nil && !os.IsNotExist(err) {
+		t.Fatalf("error checking for pre-apply dir: %v", err)
+	}
+	if os.IsNotExist(err) || !fi.IsDir() {
+		return
+	}
+	t.Log("Applying resources in pre-apply directory")
+	cmd := exec.Command("kubectl", "apply", "-f", preApplyDir)
```

Comment:
This is more of a note to self: It will be good to not depend on kubectl for just applying resources. `kpt live apply` should support this mode with a flag or something.

Reply:
Using kubectl here is useful, since it allows us to set up specific scenarios for testing that might be difficult to do using kpt itself.